### PR TITLE
[AIP-142] Fix incorrect flagging of civil dates and times (#463)

### PR DESCRIPTION
**Pull request by odsod**

### CODE REVIEW

**Comment by odsod** [comment]
ETA on CLA is ~24h. @lukesneeringer what do you think about the change?

**Comment by jgeewax** [comment]
@odsod : Can you make sure the SLA issue is resolved?

**Comment by odsod** [comment]
@jgeewax Thanks for reviewing! Yes, I'm on it.

**Comment by odsod** [comment]
@googlebot I signed it!

File: rules/aip0142/time_field_type.go [hunk 1]
```diff
@@ -27,13 +27,23 @@ import (
...
+			return nil
+		}
 		suffixes := stringset.New(
 			"date", "datetime", "ms", "msec", "msecs", "millis", "nanos", "ns",
 			"nsec", "nsecs", "sec", "secs", "seconds", "time", "timestamp", "us",
 			"usec", "usecs",
 		)
-		tokens := strings.Split(f.GetName(), "_")
-		if suffixes.Contains(tokens[len(tokens)-1]) && utils.GetTypeName(f) != "google.protobuf.Timestamp" {
+		if suffixes.Contains(suffix) && typeName != "google.protobuf.Timestamp" {
```
Comment (lukesneeringer) [inline rules/aip0142/time_field_type.go]:
```suggestion
var allowed := stringset.New(
  "google.protobuf.Timestamp",
  "google.type.Date",
  "google.type.DateTime",
  "google.type.TimeOfDay",
)
if suffixes.Contains(suffix) && !allowed.Contains(typeName) {
```

...and then remove the switch statement.

Comment (odsod) [reply rules/aip0142/time_field_type.go]: Good point! Fixed

**Comment by odsod** [comment]
@googlebot I signed it!

The CLA is signed by my CTO and the contributors group is google-contributors@einride.tech - I'm a member there with the email oscar.soderlund@einride.tech - I amended my original commit to use that email as author.

Do you need anything more from me to ensure the CLA is covered?

**Comment by odsod** [comment]
@googlebot I signed it!

**Pull request merged**